### PR TITLE
Update pre-commit and remove numpy dependency

### PR DESCRIPTION
**Pull request by MichaIng**
numpy seems to be used only to loop through a range of integers, which can be done with the builtin "range()" instead.

### CODE REVIEW

**Comment by MichaIng** [comment]
The Docker build is unrelated. It fails since the latest pillow pre-compiled wheel failed to build on piwheels: https://github.com/piwheels/packages/issues/296
So it currently is compiled, which fails as of missing build dependencies.

However, for ARMv6 and ARMv7 a large number of build dependencies would need to be added if we didn't count on piwheels, so I'd ignore this, it is usually solved quickly.

**Comment by MichaIng** [comment]
> TypeError: 'float' object cannot be interpreted as an integer

Okay `ny` is a float, which doesn't make much sense given that it is the "number of rectangles". A simply float => int conversion should do.

File: motioneye/utils/__init__.py [hunk 1]
```diff
@@ -474,9 +473,9 @@ def build_editable_mask_file(
     im = Image.new('L', (width, height), 255)  # all white
     dr = ImageDraw.Draw(im)
 
-    for y in numpy.arange(ny):
```
Comment (cclauss) [inline motioneye/utils/__init__.py:477]: `numpy.arange()` is a parallel / vector operation while Python's built-in `range()` is a serial operation but I doubt that matters unless performance is vital.

https://realpython.com/how-to-use-numpy-arange/#comparison-of-range-and-nparange

**Review by cclauss — APPROVED** [review]

File: motioneye/utils/__init__.py (hunk 1 above)
Comment (MichaIng) [reply motioneye/utils/__init__.py:477]: Thanks for the hint. I think it is not critical here, this is only done once when editing the mask and once when saving the mask, while on actual motion detection generated `/etc/motioneye/mask_X_privacy.pgm` is used as overlay. But I'll add some debug code locally to verify this.

**Comment by MichaIng** [comment]
Okay, so first of all the issue shown on above screenshot is definitely unrelated to this PR. I tested again on a fresh install on fresh VM. I'll try to solve this in a dedicated PR: #2404

The mask is parsed (including the rectangle loop) two times on every page load:
```
Apr 06 17:05:06 VM-Bullseye meyectl[4459]:     INFO: parsing editable mask privacy for camera with id 1: /etc/motioneye/mask_1.pgm
Apr 06 17:05:06 VM-Bullseye meyectl[4459]:    ERROR: failed to read mask file /etc/motioneye/mask_1.pgm: [Errno 2] No such file or directory: '/etc/motioneye/mask_1.pgm'
Apr 06 17:05:06 VM-Bullseye meyectl[4459]:     INFO: parsing editable mask privacy for camera with id 1: /etc/motioneye/mask_1.pgm
Apr 06 17:05:06 VM-Bullseye meyectl[4459]:    ERROR: failed to read mask file /etc/motioneye/mask_1.pgm: [Errno 2] No such file or directory: '/etc/motioneye/mask_1.pgm'
```
Two other things we may want to hence: Calling the function once only, and skip trying to read `mask_1.pgm` since the file is called `mask_1_privacy.pgm`.

The mask build is done only once when saving a changed mask. But while I did no measurement, I do not recognise any performance difference on page reload or saving a mask. So the loop serialisation seems to be negligible compared to fontend, network and other backend tasks.

And not loading numpy (including used libraries) on every utils usage has some (whether recognisable or not) performance and memory usage benefit on other ends 🙂.

I'll merge this hence.

**Pull request merged**